pkg/app/server.go

```diff
@@ -85,11 +87,10 @@ func RunKubeStateMetricsWrapper(ctx context.Context, opts *options.Options, fact
 // RunKubeStateMetrics will build and run the kube-state-metrics.
 // Any out-of-tree custom resource metrics could be registered by newing a registry factory
 // which implements customresource.RegistryFactory and pass all factories into this function.
-func RunKubeStateMetrics(ctx context.Context, opts *options.Options, factories ...customresource.RegistryFactory) error {
```

Comment:
Hi @mrueg, I found that this parameter `factories ...customresource.RegistryFactory` is deleted from function `RunKubeStateMetrics`.

As #1644 shows, this parameter is for some repository which does not use custom resource config file and wants to add their own metrics collecting logic in their codes. So a user could construct a new repository and vendor KSM as a dependency in `go.mod`. I believe some users have already built their project like this, e.g. https://github.com/kubernetes/kube-state-metrics/pull/1644#issuecomment-990892061.

So do you have any ideas about this scenario?